jerry-core/parser/js/js-parser-module.c

```diff
@@ -0,0 +1,803 @@
+    parser_raise_error (context_p, PARSER_ERR_NOT_IMPLEMENTED);
+  }
+
+  bool has_import_name = false;
+  lexer_literal_t *import_name_p = NULL;
+  lexer_literal_t *local_name_p = NULL;
+
+  while (true)
+  {
+    bool whole_module_needed = context_p->token.type == LEXER_MULTIPLY;
+    if ((!whole_module_needed || !(whole_module_needed && !has_import_name))
+        && (context_p->token.type != LEXER_LITERAL
+          || lexer_compare_raw_identifier_to_current (context_p, "from", 4)
+          || lexer_compare_raw_identifier_to_current (context_p, "as", 2)))
+    {
+      parser_raise_error (context_p, PARSER_ERR_INVALID_CHARACTER);
+    }
+
+    if (whole_module_needed)
+    {
+      local_name_p = NULL;
+      goto check_next_token;
+    }
+
+    lexer_construct_literal_object (context_p, &context_p->token.lit_location, LEXER_IDENT_LITERAL);
+    if (has_import_name)
+    {
+      import_name_p = context_p->lit_object.literal_p;
+    }
+    else
+    {
+      local_name_p = context_p->lit_object.literal_p;
+      import_name_p = context_p->lit_object.literal_p;
+    }
+
+    check_next_token:
+    lexer_next_token (context_p);
+
+    if (context_p->token.type == LEXER_RIGHT_BRACE
+        || (context_p->token.type == LEXER_LITERAL
+        && lexer_compare_raw_identifier_to_current (context_p, "from", 4)))
+    {
+      parser_module_add_item_to_node (context_p, module_node_p, import_name_p, local_name_p, true);
+      break;
+    }
+
+    if (context_p->token.type == LEXER_COMMA)
+    {
+      parser_module_add_item_to_node (context_p, module_node_p, import_name_p, local_name_p, true);
+      has_import_name = false;
+    }
+    else if (context_p->token.type == LEXER_LITERAL
+             && lexer_compare_raw_identifier_to_current (context_p, "as", 2))
+    {
+      if (has_import_name)
+      {
+        parser_raise_error (context_p, PARSER_ERR_INVALID_CHARACTER);
+      }
+
+      has_import_name = true;
+    }
+    else
+    {
+      parser_raise_error (context_p, PARSER_ERR_RIGHT_PAREN_COMMA_FROM_EXPECTED);
+    }
+    lexer_next_token (context_p);
+  }
+} /* parser_module_parse_import_item_list */
+
+/**
+ * Check if property is exported from the script.
+ * @returns true - if the given property is exported
+ *          false - otherwise
+ */
+static bool
+parser_module_is_property_exported (ecma_string_t *property_name_p, /**< property name */
+                                    parser_module_node_t *export_node_p, /**< export node */
+                                    parser_module_names_t *export_name_p) /**< export name */
+{
+  parser_module_names_t *current = export_node_p->module_names_p;
+  parser_module_names_t *next;
+
+  if (current == NULL)
+  {
+    return false;
+  }
+
+  for (int i = 0; i < export_node_p->module_request_count; i++)
+  {
+    next = current->next_p;
+    ecma_string_t *import_name_p = ecma_new_ecma_string_from_utf8 (current->local_name_p,
```

Comment:
All these string can be determined during parse time, therefor ecma_find_or_create_literal can be used here. (Maybe some modifications are required)

Reply:
The first row of that [function](https://github.com/pando-project/jerryscript/blob/master/jerry-core/ecma/base/ecma-literal-storage.c#L73) is `ecma_string_t *string_p = ecma_new_ecma_string_from_utf8 (chars_p, size);` and does a lot more (unnecessary in this case) things after it, so I think this method is not as costly as calling the `ecma_find_or_create_literal_string`.